### PR TITLE
Remove unnecessary V.KernelFormatterHandler inheritance

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117673

Why we running kernel formatting when doing a read-write analysis?
Makes no sense.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/dependencies.py [hunk 1]
```diff
@@ -356,7 +347,8 @@ def extract_read_writes(
     prefix: str = "d",
 ):
     args, var_ranges = index_vars_squeeze(*argsizes, prefix=prefix)
-    rw = RecordLoadStore(var_ranges, normalize=normalize)
+    inner = _RecordLoadStoreInner(var_ranges=var_ranges, normalize=normalize)
```
Comment (peterbell10) [inline torch/_inductor/dependencies.py:350]: `RecordLoadStoreInner` inherits from `V.MockHandler` which also does string formatting but builds up everything in a single line with no use of varables, so the expressions can build up exponentially. e.g.
```python
b = a * a
c = b * b
d = c * c
```
will be represented as the string
`ops.mul(ops.mul(ops.mul(a, a), ops.mul(a, a)), ops.mul(a, a), ops.mul(a, a))), ops.mul(ops.mul(a, a), ops.mul(a, a)), ops.mul(a, a), ops.mul(a, a)))`


A better solution would be to have another base that returns `tmp{n}` without actually building up a string at all.

**Review by peterbell10 — CHANGES_REQUESTED** [review]

File: torch/_inductor/dependencies.py (hunk 1 above)
Comment (ezyang) [reply torch/_inductor/dependencies.py:350]: I think that RecordLoadStoreInner doesn't even look at the returned outputs at all, and I was planning on getting rid of the MockHandler inheritance, so I guess I'll do it all in one go.

Comment (peterbell10) [reply torch/_inductor/dependencies.py:350]: I believe the only cases where it matters is when the return type isn't directly an IR value, so `index_expr` must return a sympy expression and `reduction` must sometimes return a tuple of IR values. What exactly that IR value is represented as doesn't matter.

**Comment by ezyang** [comment]
Obsoleted by https://github.com/pytorch/pytorch/pull/117679

**Pull request closed without merging**